src/main/java/taplink/network/menu/api/service/impl/StoreServiceImpl.java

```diff
@@ -31,5 +41,25 @@ public ResponseDto<StoreResponseDto> searchStores(String searchKey, int pageNo,
         return new ResponseDto<>(stores, content);
     }
 
+    @Override
+    public StoreResponseDto createStore(String storeJson, MultipartFile image) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            StoreRequestDto storeRequestDto = objectMapper.readValue(storeJson, StoreRequestDto.class);
+            Store store = objectMapperUtils.convertEntityAndDto(storeRequestDto, Store.class);
+            Store savedStore = storeRepository.save(store);
+            return objectMapperUtils.convertEntityAndDto(savedStore, StoreResponseDto.class);
+        } catch (IOException ex) {
+            logger.error("Getting exception while reading storeJson", ex);
+            throw new RuntimeException(ex);
+        }
+    }
+
+    @Override
+    public StoreResponseDto findById(Long id) {
+        Store store = storeRepository.findById(id).orElseThrow(() -> new ResourceNotFoundException("Store not found for id: " + id));
```

Comment:
Can we create a class constant for this error message?

Reply:
Create a class, and put them to class

Reply:
Why do we need to create a constant for this message, This is a message for a specific error, we will not reuse this anywhere

Reply:
But we can put the generic message like {Entity} + could not be found for {id} into the Exception class. Then we can pass specific entity name and the Id to the constructor when we throw the exception